src/Authorization/Helpers/PolicyHelper.cs

```diff
@@ -51,6 +51,38 @@ public static List<string> GetRolesWithAccess(XacmlPolicy policy)
             return roleCodes.ToList();
         }
 
+        /// <summary>
+        /// Returns a list of uniue subject attribute ids. Used to identify the role/group and other sources needed to enirch subject info
+        /// </summary>
+        /// <param name="policy">The policy</param>
+        /// <returns>List of attribute ids</returns>
+        public static List<string> GetSubjectAttributeIds(XacmlPolicy policy)
```

Comment:
A possible improvement could be to cache this list together with the policy. I am not sure if this iteration might be costly if it happens enough

Reply:
Good point! Will look into it

Reply:
Moving the implementation to the PRP and caching it with the policy feels like the wrong place.. and will need changes to the interface and existing use of the operations.

Ended up wrapping the use of the operation from ContextHandler in a GetFromCacheOrPolicy helper method.
But it then also required getting a PolicyId to be able to cache the result..

Reply:
Moved as a helper function on the XacmlPolicy implementation in ABAC to extract all attributes for a given category